### PR TITLE
trt-1892: check for pod security audit violations

### PR DESCRIPTION
**Pull request by neisw**

### CODE REVIEW

**Comment by neisw** [comment]
/payload-job periodic-ci-openshift-frr-release-4.19-periodics-frrk8s-e2e-metal-cno-periodic

**Comment by neisw** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.19-e2e-aws-ovn-techpreview-serial

**Comment by neisw** [comment]
/retitle trt-1892: check for pod security audit violations

File: pkg/monitortests/kubeapiserver/auditloganalyzer/handle_audit_violation.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	"strings"
+	"sync"
+)
+
+func CheckForViolations() *violations {
+	return &violations{}
+}
+
+type violations struct {
```
Comment (sosiouxme) [inline pkg/monitortests/kubeapiserver/auditloganalyzer/handle_audit_violation.go:16]: normally i'm biased toward shorter names but in this case... "violation" seems a little raw by itself. maybe `auditViolations` and similar for other names?

File: pkg/monitortests/kubeapiserver/auditloganalyzer/handle_audit_violation.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	case len(v.records) > 0:
+		messages := []string{}
+		for _, v := range v.records {
+			messages = append(messages, fmt.Sprintf("%s: %s %s/%s: %s - %s", v.auditId, v.resource, v.namespace, v.name, v.username, v.violation))
+		}
+		ret = append(ret,
+			&junitapi.JUnitTestCase{
+				Name: testName,
+				FailureOutput: &junitapi.FailureOutput{
+					Message: fmt.Sprintf("%s", strings.Join(messages, "\n")),
```
Comment (sosiouxme) [inline pkg/monitortests/kubeapiserver/auditloganalyzer/handle_audit_violation.go:64]: why not just...
```suggestion
					Message: strings.Join(messages, "\n"),
```

Comment (neisw) [reply pkg/monitortests/kubeapiserver/auditloganalyzer/handle_audit_violation.go:64]: Because the code I copied had a prefix there...

**Pull request merged**